src/lib.rs

```diff
@@ -186,19 +186,61 @@ pub fn to_optional_default<'a, T: PartialEq>(v: &'a T, default: &'a T) -> Option
 /// This API is public so that it may be used from macros, but should not be
 /// considered a part of the supported API surface.
 #[doc(hidden)]
-pub const fn implicit_tag(tag: u32, inner_tag: Tag) -> Tag {
+pub const fn implicit_tag_class<const TAG_CLASS: u8>(tag: u32, inner_tag: Tag) -> Tag {
     Tag::new(
         tag,
-        tag::TagClass::ContextSpecific,
+        TagClass::from_u8(TAG_CLASS),
         inner_tag.is_constructed(),
     )
 }
 
+/// This API is public so that it may be used from macros, but should not be
+/// considered a part of the supported API surface.
+#[doc(hidden)]
+pub const fn implicit_tag(tag: u32, inner_tag: Tag) -> Tag {
+    implicit_tag_context_specific(tag, inner_tag)
+}
+
+/// This API is public so that it may be used from macros, but should not be
+/// considered a part of the supported API surface.
+#[doc(hidden)]
+pub const fn implicit_tag_application(tag: u32, inner_tag: Tag) -> Tag {
+    Tag::new(tag, TagClass::Application, inner_tag.is_constructed())
+}
+
+/// This API is public so that it may be used from macros, but should not be
+/// considered a part of the supported API surface.
+#[doc(hidden)]
+pub const fn implicit_tag_context_specific(tag: u32, inner_tag: Tag) -> Tag {
```

Comment:
No, true - I also removed the u8 from the generic parameters and changed the signature like so:
```rust
implicit_tag_class(tag: u8, tag_class: TagClass, ..)
```
adding the TagClass as parameter instead of a const-generic u8